src/Service/FileService.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Service;
+
+use App\Service\Interface\FileServiceInterface;
+use League\Flysystem\FilesystemException;
+use League\Flysystem\FilesystemOperator;
+use League\Flysystem\UnableToDeleteFile;
+use League\Flysystem\UnableToWriteFile;
+
+class FileService implements FileServiceInterface
+{
+    private FilesystemOperator $filesystem;
+
+    public function __construct(FilesystemOperator $defaultStorage)
+    {
+        $this->filesystem = $defaultStorage;
+    }
```

Comment:
acho que rola utilizar a `constructor promotion property` https://stitcher.io/blog/constructor-promotion-in-php-8

```php
public function __construct(
  private FilesystemOperator $filesystem
) { 
}
```